react/week2/todolist/src/App.js

```diff
@@ -0,0 +1,95 @@
+import './App.css';
+import React, { useEffect, useState } from "react";
+
+const todos = [
+    {
+        id: 1,
+        description: "Get out of bed",
+    },
+    {
+        id: 2,
+        description: "Brush teeth",
+    },
+    {
+        id: 3,
+        description: "Eat breakfast",
+    },
+];
+
+const myTodoList = ["Finish my hyf homework", "Walk the dog", "Do laundry", "Prepare for class"]
+
+function TodoItem(props) {
+    const [todoStatus, setTodoStatus] = useState(true);
+    function checkState() {
+        setTodoStatus(!todoStatus)
+    }
+
+    return (
+        <div className="todo-item">
+            <div className="item-description">
+                <input type="checkbox" onClick={checkState}></input>
+                <li style={{ 'textDecoration': todoStatus ? 'none' : 'line-through' }} >{props.description}</li>
+            </div>
+            <div>
+                <button onClick={() => props.deleteMethod(props.id)}> Delete </button>
+            </div>
+        </div>
+    )
+}
+
+function ListContainer() {
+    return (
+        <div className="list-container">
+            <h1>Todo List</h1>
+            <List />
+        </div>
+    )
+}
+
+function List() {
+    const [todo, setTodo] = useState(todos)
+    function addTodo() {
+        const randomIndex = Math.floor(Math.random() * myTodoList.length);
+        setTodo(prev => [...prev, {
+            id: prev.length + 1,
+            description: myTodoList[randomIndex]
+        }]);
+    }
+    function deleteTodo(itemID) {
+        setTodo((prev) => {
+            return prev.filter((item, index) => {
+                return index !== itemID
+            });
+        })
+    }
+    return (
+        <div>
+            {todo.length === 0 && <p>No items</p>}
+            <ul>
+                {todo.map((item, index) => <TodoItem key={item.id} description={item.description} deleteMethod={deleteTodo} id={index} />)}
```

Comment:
Here you pass the index as the id. This is part of the reason why your functionality isn't working

Reply:
It worked, I was filtering by the array index so it was working. I changed it to the item.id and is also working. I guess the item.id is better for what you told us in class, right?

Reply:
Correct yes. I see now that it is working. I was thrown off because you passed the index but called it itemID in the delete method. So yes, if you fix setting the id in the add method, it will work.
I would still prefer that you delete by id, though. Imagine for instance you have to make a db call when you delete.